[Actions][Docs] Modify connector documentation with the new MS Exchange online configuration requirements (#112768)
elastic/kibana: docs/management/connectors/action-types/email.asciidoc

<titleabbrev>Email</titleabbrev>
++++

> The email connector uses the SMTP protocol to send mail messages, using an integration of https://nodemailer.com/[Nodemailer]. An exception is Microsoft Exchange, which uses HTTP protocol for sending emails https://docs.microsoft.com/en-us/graph/api/user-sendmail[Send mail]. Email message text is sent as both plain text and html text.

Reviewer: ```suggestion
The email connector uses the SMTP protocol to send mail messages, using an integration of https://nodemailer.com/[Nodemailer]. An exception is Microsoft Exchange, which uses HTTP protocol for sending emails, https://docs.microsoft.com/en-us/graph/api/user-sendmail[Send mail]. Email message text is sent as both plain text and html text.
```

> The email connector uses the SMTP protocol to send mail messages, using an integration of https://nodemailer.com/[Nodemailer], except Microsoft Exchange, which use its own integration using https://docs.microsoft.com/en-us/graph/api/overview[Graph API] method https://docs.microsoft.com/en-us/graph/api/user-sendmail[Send mail] on the top of HTTP protocol. Email message text is sent as both plain text and html text.

Reviewer: Can you break this into two sentences:

The email connector uses the SMTP protocol to send mail messages, using an integration of Nodemailer.  An exception is Microsoft Exchange, which uses the Send mail method in the Graph API on the top of HTTP protocol.

Not sure what this means:

using Graph API method Send mail on the top of HTTP protocol

Name::      The name of the connector. The name is used to identify a  connector in the management UI connector listing, or in the connector list when configuring an action.
Sender::    The from address for all emails sent with this connector. This can be specified in `user@host-name` format or as `"human name <user@host-name>"` format. See the https://nodemailer.com/message/addresses/[Nodemailer address documentation] for more information.

> Service::   The name of the email service. If `service` is one of Nodemailer's https://nodemailer.com/smtp/well-known/[well-known email service providers], `host`, `port`, and `secure` properties will be defined with the default values and disabled for modification. If `service` is `MS Exchange Server` the `host`, `port`, and `secure` properties will be ignored and `tenantId`, `clientId`, `clientSecret` will be required instead. If `service` is `Other`, the `host` and `port` properties must be defined.

Reviewer: ```suggestion
Service::   The name of the email service. If `service` is one of Nodemailer's https://nodemailer.com/smtp/well-known/[well-known email service providers], the `host`, `port`, and `secure` properties are defined with the default values and disabled for modification. If `service` is `MS Exchange Server`, the `host`, `port`, and `secure` properties are ignored and `tenantId`, `clientId`, `clientSecret` are required instead. If `service` is `other`, the `host` and `port` properties must be defined.
```

Host::      Host name of the service provider. If you are using the <<action-settings, `xpack.actions.allowedHosts`>> setting, make sure this hostname is added to the allowed hosts.
Port::      The port to connect to on the service provider.
Secure::    If true, the connection will use TLS when connecting to the service provider. Refer to the https://nodemailer.com/smtp/#tls-options[Nodemailer TLS documentation] for more information.  If not true, the connection will initially connect over TCP, then attempt to switch to TLS via the SMTP STARTTLS command.

> Tenant ID:: The directory tenant the application plans to operate against, in GUID format.

Reviewer: ```suggestion
Tenant ID:: The directory tenant that the application plans to operate against, in GUID format.
```

> Client ID:: The application ID that's assigned to your app, in GUID format. You can find this information in the portal where you registered your app.

Reviewer: ```suggestion
Client ID:: The application ID that is assigned to your app, in GUID format. You can find this information in the portal where you registered your app.
```

> Client Secret:: The client secret that you generated for your app in the app registration portal. The client secret must be URL-encoded before being sent. The Basic auth pattern of instead providing credentials in the Authorization header, per https://datatracker.ietf.org/doc/html/rfc6749#section-2.3.1[RFC 6749] is also supported.

Reviewer: ```suggestion
Client Secret:: The client secret that you generated for your app in the app registration portal. The client secret must be URL-encoded before being sent. The Basic auth pattern of providing credentials in the Authorization header, per https://datatracker.ietf.org/doc/html/rfc6749#section-2.3.1[RFC 6749], is also supported.
```

Config defines information for the connector type.

> `service`:: The name of the email service. If `service` is `elastic_cloud` (for Elastic Cloud notifications) or one of Nodemailer's https://nodemailer.com/smtp/well-known/[well-known email service providers], `host`, `port`, and `secure` properties are ignored. If `service` is `other`, `host` and `port` properties must be defined. For more information on the `gmail` service value, see the https://nodemailer.com/usage/using-gmail/[Nodemailer Gmail documentation]. If `service` is `exchange_server`, the `tenantId`, `clientId`, `clientSecret` properties will be required instead of `host` and `port`.

Reviewer: ```suggestion
`service`:: The name of the email service. If `service` is `elastic_cloud` (for Elastic Cloud notifications) or one of Nodemailer's https://nodemailer.com/smtp/well-known/[well-known email service providers], the `host`, `port`, and `secure` properties are ignored. If `service` is `other`,  the `host` and `port` properties must be defined. For more information on the `gmail` service value, refer to https://nodemailer.com/usage/using-gmail/[Nodemailer Gmail documentation]. If `service` is `exchange_server`, the `tenantId`, `clientId`, `clientSecret` properties are required instead of `host` and `port`.
```

`from`:: An email address that corresponds to *Sender*.
`host`:: A string that corresponds to *Host*.
`port`:: A number that corresponds to *Port*.
`secure`:: A boolean that corresponds to *Secure*.
`hasAuth`:: A boolean that corresponds to *Requires authentication*. If `true`, this connector will require values for `user` and `password` inside the secrets configuration. Defaults to `true`.

> `tenantId`:: A GUID format value that corresponds *Tenant ID*, which is a part of OAuth 2.0 Client Credentials Authentication. 

Reviewer: ```suggestion
`tenantId`:: A GUID format value that corresponds to *Tenant ID*, which is a part of OAuth 2.0 Client Credentials Authentication. 
```

> `clientId`:: A GUID format value that corresponds *Client ID*, which is a part of OAuth 2.0 Client Credentials Authentication.

Reviewer: ```suggestion
`clientId`:: A GUID format value that corresponds to *Client ID*, which is a part of OAuth 2.0 Client Credentials Authentication.
```

[[configuring-email]]
==== Configuring email accounts for well-known services

> The email connector can send email using many popular SMTP email services and Microsoft Exchange Graph API.

Reviewer: ```suggestion
The email connector can send email using many popular SMTP email services and the Microsoft Exchange Graph API.
```

[source,text]
--------------------------------------------------
  config:
    service: gmail

>     // `host`, `port` and `secure` have default values below and no need to set: 

Reviewer: ```suggestion
    // `host`, `port` and `secure` have the following default values and do not need to set: 
```

[source,text]
--------------------------------------------------
config:
    service: outlook365

>     // `host`, `port` and `secure` have default values below and no need to set: 

Reviewer: ```suggestion
    // `host`, `port` and `secure` have the following default values and do not need to set: 
```

[source,text]
--------------------------------------------------
config:
    service: ses

>     // `host`, `port` and `secure` have default values below and no need to set: 

Reviewer: ```suggestion
    // `host`, `port` and `secure` have the following default values and do not need to set: 
```

[[exchange]]
==== Sending email from Microsoft Exchange

> deprecated:[This Microsoft Exchange configuration is deprecated in 7.16.0, and will be removed later, because of the Microsoft announcement about deprecating https://docs.microsoft.com/en-us/lifecycle/announcements/exchange-online-basic-auth-deprecated [Basic Authentication]. To prepare for the removal of Basic Auth, you must update all existing Microsoft Exchange connectors with the new configuration based on the https://docs.microsoft.com/en-us/azure/active-directory/develop/v2-oauth2-client-creds-grant-flow[OAuth 2.0 Client Credentials Authentication].

Reviewer: Can the first line be shortened:

This Microsoft Exchange configuration is deprecated in 7.16.0, and will be removed later, because Microsoft is deprecating Basic Authentication.

<2> Many organizations support use of your email address as your username.
    Check with your system administrator if you receive
    authentication-related failures.

> 

Reviewer 2: I think we need a section header here to identify that this is how to set up the new configuration for MS Exchange. I was confused when I first read through because it talked about the deprecated config but then moved on to the new config without a transition.

> Before you create an email connector for Microsoft Exchange, you must create and register the client integration application on the https://go.microsoft.com/fwlink/?linkid=2083908[Azure portal]:

Reviewer 2: I think we need a section header here to identify that this is how to set up the new configuration for MS Exchange. I was confused when I first read through because it talked about the deprecated config but then moved on to the new config without a transition.

[role="screenshot"]
image::management/connectors/images/exchange-api-permissions.png[MS Exchange API permissions]

Add the "Mail.Send" permission for Microsoft Graph.  The permission appears in the list with the status "Not granted for <your Azure active directory>":
[role="screenshot"]
image::management/connectors/images/exchange-not-granted.png[MS Exchange "Mail.Send" not granted]

Click *Grant admin consent for <your Azure active directory>*.
[role="screenshot"]
image::management/connectors/images/exchange-grant-confirm.png[MS Exchange grant confirmation]

Confirm that the status for the "Mail.Send" permission is now granted.
[role="screenshot"]
image::management/connectors/images/exchange-granted.png[MS Exchange grant confirmation]

[float]
[[exchange-client-secret]]
===== Configure Microsoft Exchange Client secret
To configure the Client secret , open *Manage > Certificates & secrets*.
[role="screenshot"]
image::management/connectors/images/exchange-secrets.png[MS Exchange secrets configuration]

> Add a new client secret, then copy the value and put it to the proper field in the Microsoft Exchange email connector.

Author: I think we have a bunch now, probably when the SDH comes :-)

<2> Many organizations support use of your email address as your username.
    Check with your system administrator if you receive
    authentication-related failures.

> Before create the email connector for Microsoft Exchange, you need to create and register the client integration application on the https://go.microsoft.com/fwlink/?linkid=2083908[Azure portal]:

Reviewer: ```suggestion
Before you create an email connector for Microsoft Exchange, you must create and register the client integration application on the https://go.microsoft.com/fwlink/?linkid=2083908[Azure portal]:
```

[role="screenshot"]
image::management/connectors/images/exchange-register-app.png[Register client application for MS Exchange]

> The next step is defigning API permissions for the registered application to be able to send emails. https://docs.microsoft.com/en-us/graph/api/user-sendmail?view=graph-rest-1.0&tabs=http#permissions[Documentation] for Microsoft Graph API.

Reviewer: ```suggestion
On the Azure portal, open *Manage > API permissions*, and then define the permissions for the registered application to send emails. Refer to the https://docs.microsoft.com/en-us/graph/api/user-sendmail?view=graph-rest-1.0&tabs=http#permissions[documentation] for the Microsoft Graph API.
```

> On the Azure portal, open *Manage > API permissions*, and then define the permissions for the registered application to send emails. Refer to the https://docs.microsoft.com/en-us/graph/api/user-sendmail?view=graph-rest-1.0&tabs=http#permissions[documentation] for the Microsoft Graph API.

Reviewer: ```suggestion
Next, open *Manage > API permissions*, and then define the permissions for the registered application to send emails. Refer to the https://docs.microsoft.com/en-us/graph/api/user-sendmail?view=graph-rest-1.0&tabs=http#permissions[documentation] for the Microsoft Graph API.
```

> You can configure it on the Azure portal application page, Manage - API permissions:

Reviewer: remove this sentence.  the info is in the first sentence above.

[role="screenshot"]
image::management/connectors/images/exchange-api-permissions.png[MS Exchange API permissions]

> The minimum requirenment is to add "Mail.Send" permission for Microsoft Graph and it will appear in the list with the status "Not granted for <Your Azure Active Directory>":

Reviewer: ```suggestion
Add the "Mail.Send" permission for Microsoft Graph.  The permission appears in the list with the status "Not granted for <your Azure active directory>":
```

[role="screenshot"]
image::management/connectors/images/exchange-not-granted.png[MS Exchange "Mail.Send" not granted]

> To make this permission working as expected you need to grant admin consent for you Azure Active Directory by clicking on the button "Grant admin consent for <Your Azure Active Directory>":

Reviewer: ```suggestion
Click *Grant admin consent for <your Azure active directory>*.
```

[role="screenshot"]
image::management/connectors/images/exchange-grant-confirm.png[MS Exchange grant confirmation]

> Confirm the grant. The status for the "Mail.Send" permission changes to granted.

Reviewer: should this be:

Confirm that the status for the "Mail.Send" permission is now granted

> After confirming the grant, "Mail.Send" permission will be displayed as granted:

Reviewer: ```suggestion
Confirm the grant. The status for the "Mail.Send" permission changes to granted.
```

[role="screenshot"]
image::management/connectors/images/exchange-granted.png[MS Exchange grant confirmation]

[float]
[[exchange-client-secret]]
===== Configure Microsoft Exchange Client secret

> To configure the Client secret , open *Manage > Certificates & secrets* on the Azure portal.

Reviewer: ```suggestion
To configure the Client secret , open *Manage > Certificates & secrets*.
```

> To configure Client secret you should use Azure portal application Manage - Certificates & secrets:

Reviewer: ```suggestion
To configure the Client secret , open *Manage > Certificates & secrets* on the Azure portal.
```

[role="screenshot"]
image::management/connectors/images/exchange-secrets.png[MS Exchange secrets configuration]

> Add a new client secret, then copy the value and put it to the proper field in Microsoft Exchange email connector.

Reviewer: ```suggestion
Add a new client secret, then copy the value and put it to the proper field in the Microsoft Exchange email connector.
```

> After adding a new client secret copy the value and put it to the proper field in Microsoft Exchange email connector.

Reviewer: ```suggestion
Add a new client secret, then copy the value and put it to the proper field in Microsoft Exchange email connector.
```

[float]
[[exchange-client-tenant-id]]
===== Configure Microsoft Exchange Client ID and Tenant ID

> To find application Client ID,  open the *Overview* page on the Azure portal.

Reviewer: ```suggestion
To find the application Client ID,  open the *Overview* page.
```

> To find application Client ID you should use Azure portal application Overview page:

Reviewer: ```suggestion
To find application Client ID,  open the *Overview* page on the Azure portal.
```

[role="screenshot"]
image::management/connectors/images/exchange-client-tenant.png[MS Exchange Client ID and Tenant ID configuration]

> Copy and paste this values to the proper fields in Microsoft Exchange email connector.

Reviewer: ```suggestion
Copy and paste this values to the proper fields in the Microsoft Exchange email connector.
```

> The last step will use the following email connector configuration to send email from Microsoft Exchange:

Reviewer: ```suggestion
Use the following email connector configuration to send email from Microsoft Exchange:
```

[source,text]
--------------------------------------------------
config:
    service: exchange_server
    clientId: <The Application (client) ID> <1>
    tenantId: <The directory tenant ID, in GUID format.>
    from: <email address of service account> <2>
secrets:
    clientSecret: <URL-encoded string>
--------------------------------------------------

> <1> This applicatin information could be find by on the https://go.microsoft.com/fwlink/?linkid=2083908[Azure portal – App registrations].

Reviewer: ```suggestion
<1> This application information is on the https://go.microsoft.com/fwlink/?linkid=2083908[Azure portal – App registrations].
```
